### PR TITLE
cnf-tests: Delete macvlan nad to retry allowlist test

### PR DESCRIPTION
**Pull request by mlguerrero12**
In case of failure, subsequent retries of the allowlist test fail because the macvlan is already present. This changes ensure that the macvlan nad is deleted.

### CODE REVIEW

File: cnf-tests/testsuites/e2esuite/security/tuning.go [hunk 1]
```diff
@@ -152,16 +160,19 @@ var _ = Describe("[tuningcni]", func() {
 		})
 
 		AfterEach(func() {
-			updateAllowlistConfig(originalSysctls)
+			err := client.Client.Delete(context.Background(), macVlandNad)
```
Comment (zeeke) [inline cnf-tests/testsuites/e2esuite/security/tuning.go:163]: Does it panic if `macVlandNad` is nil? What about using `DeferCleanup(...)` after line 176?

Comment (mlguerrero12) [reply cnf-tests/testsuites/e2esuite/security/tuning.go:163]: I'm not sure if it panics with nil. Nevertheless, thanks for the suggestion. I'll use `DeferCleanup`.

File: cnf-tests/testsuites/e2esuite/security/tuning.go [hunk 2]
```diff
@@ -166,6 +162,8 @@ var _ = Describe("[tuningcni]", func() {
 			err = client.Client.Create(context.Background(), macVlandNad)
 			Expect(err).ToNot(HaveOccurred())
 
+			DeferCleanup(client.Client.Delete, context.Background(), macVlandNad)
```
Comment (zeeke) [inline cnf-tests/testsuites/e2esuite/security/tuning.go:165]: nice!!

**Comment by SchSeba** [comment]
failed test is not related
/override ci/prow/e2e-aws-ci-tests
/lgtm
/approve

**Pull request merged**